### PR TITLE
Specify max bytes per filter string and max bytes per aggregation key identifier as constants

### PR DESCRIPTION
**Pull request by linnan-github**
***
<a href="https://pr-preview.s3.amazonaws.com/linnan-github/conversion-measurement-api/pull/1066.html" title="Last updated on Oct 16, 2023, 1:47 PM UTC (d4c010c)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/WICG/attribution-reporting-api/1066/6198a8c...linnan-github:d4c010c.html" title="Last updated on Oct 16, 2023, 1:47 PM UTC (d4c010c)">Diff</a>

### CODE REVIEW

File: index.bs
Comment (apasel422) [inline index.bs:1200]:
```suggestion
<dfn>Max bytes per filter string</dfn> is a positive integer that controls the
```

**Review by apasel422 — CHANGES_REQUESTED** [review]
I'm not completely sure that the spec changes here match the currently implemented behavior in Chromium, based on [the Infra definition of a string](https://infra.spec.whatwg.org/#strings), which "is a sequence of unsigned 16-bit integers", and the TODO at https://wicg.github.io/attribution-reporting-api/#issue-e6d5a493.

I think it's also a bit confusing for our spec to use the term `bytes` when Infra strings do not work that way.

Might be good to get an expert in this area to take a look, as well as adding a non-normative note to the spec explaining why these limits exist in the first place.

**Comment by linnan-github** [comment]
> I'm not completely sure that the spec changes here match the currently implemented behavior in Chromium, based on [the Infra definition of a string](https://infra.spec.whatwg.org/#strings), which "is a sequence of unsigned 16-bit integers", and the TODO at https://wicg.github.io/attribution-reporting-api/#issue-e6d5a493.
> 
> I think it's also a bit confusing for our spec to use the term `bytes` when Infra strings do not work that way.
> 
> Might be good to get an expert in this area to take a look, as well as adding a non-normative note to the spec explaining why these limits exist in the first place.

Thanks. Updated to use length instead of bytes.

@yoavweiss Could you please take a look and suggest how to properly spec the string length limit check? Thanks!

**Review by apasel422 — CHANGES_REQUESTED** [review]

File: ts/src/header-validator/validate-json.ts
Comment (apasel422) [inline ts/src/header-validator/validate-json.ts:564]: It may be easier to understand the errors here with an optional prefix so that the key being too long can be more easily distinguished from the value being too long, e.g.

```ts
const filterStringLength = (s: string, errPrefix: string = '') => {
  ...
  ctx.error(`${errPrefix}exceeds...`)

...

if (!filterStringLength(key, 'key ')) {
```

Same for the aggregation keys below.

**Review by apasel422 — CHANGES_REQUESTED** [review]

File: ts/src/header-validator/validate-json.ts
Comment (linnan-github) [reply ts/src/header-validator/validate-json.ts:564]: Is that necessary with the path already?

Comment (apasel422) [reply ts/src/header-validator/validate-json.ts:564]: Yes, because the path does not contain an element that distinguishes key from value.

Comment (linnan-github) [reply ts/src/header-validator/validate-json.ts]: Updated with the error prefix, thanks!

File: ts/src/header-validator/validate-json.ts
Comment (apasel422) [inline ts/src/header-validator/validate-json.ts:1110]:
```suggestion
    string(ctx, j).filter((s) =>
      aggregationKeyIdentifierLength(ctx, s)
    )
```

**Review by apasel422 — APPROVED** [review]
Header validator LGTM. Please wait for approval of the spec changes.

**Comment by linnan-github** [comment]
@yoavweiss, just friendly ping, could you please take a look whether this is the right way to specify the limit on the string length? Thanks!!

**Comment by yoavweiss** [comment]
The current definitions based on string/length seem fine to me.

**Comment by linnan-github** [comment]
> The current definitions based on string/length seem fine to me.

Thanks @yoavweiss!

**Review by apasel422 — APPROVED** [review]

**Pull request merged**
